File: app/src/main/java/br/com/digitalhouse/projetomarvel/view/activity/DetailsActivity.java
package br.com.digitalhouse.projetomarvel.view.activity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.squareup.picasso.Picasso;

import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

import br.com.digitalhouse.projetomarvel.R;
import br.com.digitalhouse.projetomarvel.pojo.Result;
import br.com.digitalhouse.projetomarvel.view.Interface.OnClickImageDetails;

import static br.com.digitalhouse.projetomarvel.view.activity.MainActivity.RESULT_KEY;

public class DetailsActivity extends AppCompatActivity implements OnClickImageDetails {

    private ImageView imgDetails, img2Details;
    private TextView titleview, descview, publishedview, priceView, pagesview;

    public static final String DETAILSIMAGE_KEY = "imagedetails";


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_details);

        initViews();

        if (getIntent() != null && getIntent().getExtras() != null) {

            Result result = getIntent().getExtras().getParcelable(RESULT_KEY);

            carregaImagens(result);

            SetAsViews(result);

            formatDateSetText(result);

            cliqueImagem(result);
        }
    }

    private void SetAsViews(Result result) {
        titleview.setText(result.getTitle());
        descview.setText(result.getDescription());
        priceView.setText(result.getPrices().get(0).getPrice());
        pagesview.setText(result.getPageCount());
    }

    private void carregaImagens(Result result) {
        Picasso.get().load(result.getThumbnail().getPath() + ".jpg").into(imgDetails);
        Picasso.get().load(result.getThumbnail().getPath() + ".jpg").into(img2Details);

    }

    private void formatDateSetText(Result result) {
        try {
            Locale local = new Locale("pt", "BR");
            DateFormat inputFormat = new SimpleDateFormat("yyyy-MM-dd");
            DateFormat outputFormat = new SimpleDateFormat("MMMM ',' dd ',' yyyy", local);
            String inputText = result.getDates().get(0).getDate();
            Date date = inputFormat.parse(inputText);
            String outputText = outputFormat.format(date);
            publishedview.setText(outputText);

        } catch (Exception ex) {

            Log.i("Exception", "------------> " + ex.getMessage());
        }
    }

    private void cliqueImagem(Result result) {

        imgDetails.setOnClickListener(v -> {
            Intent intent = new Intent(DetailsActivity.this, DetailsImageActivity.class);
            intent.putExtra(DETAILSIMAGE_KEY, result);
            startActivity(intent);

        });
    }


    public void initViews() {

        imgDetails = findViewById(R.id.imageDetails);
        img2Details = findViewById(R.id.imagem2Details);
        titleview = findViewById(R.id.titleDetails);
        descview = findViewById(R.id.descDetails);
        publishedview = findViewById(R.id.publishedetails);
        priceView = findViewById(R.id.priceDetails);
        pagesview = findViewById(R.id.pagesDetails);


    }

    @Override
    public void Onclick(Result result) {
        Intent intent = new Intent(this, DetailsImageActivity.class);
        Bundle bundle = new Bundle();
        bundle.putParcelable(RESULT_KEY, result);
        intent.putExtras(bundle);
        startActivity(intent);
        finish();
    }
}

File: app/src/main/java/br/com/digitalhouse/projetomarvel/view/activity/DetailsActivity.kt
package br.com.digitalhouse.projetomarvel.view.activity

import android.content.Intent
import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import br.com.digitalhouse.projetomarvel.R
import br.com.digitalhouse.projetomarvel.pojo.Result
import br.com.digitalhouse.projetomarvel.view.Interface.OnClickImageDetails
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.activity_details.*
import java.text.DateFormat
import java.text.SimpleDateFormat
import java.util.*

class DetailsActivity : AppCompatActivity(), OnClickImageDetails {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_details)

        if (intent != null && intent.extras != null) {
            val result: Result? = intent.extras!!.getParcelable(MainActivity.RESULT_KEY)

            carregaImagens(result)

            SetAsViews(result)

            formatDateSetText(result)

            cliqueImagem(result)
        }
    }


    private fun SetAsViews(result: Result?) {
        titleDetails!!.text = result!!.title
        descdetails!!.text = result.description
        priceDetails!!.text = result.prices[0].price
        pagesDetails!!.text = result.pageCount
    }

    private fun carregaImagens(result: Result?) {
        Picasso.get().load(result!!.thumbnail.path + ".jpg").into(imageDetails)
        Picasso.get().load(result.thumbnail.path + ".jpg").into(imagem2Details)
    }

    private fun formatDateSetText(result: Result?) {
        try {
            val local = Locale("pt", "BR")
            val inputFormat: DateFormat = SimpleDateFormat("yyyy-MM-dd")
            val outputFormat: DateFormat = SimpleDateFormat("MMMM ',' dd ',' yyyy", local)
            val inputText = result!!.dates[0].date
            val date = inputFormat.parse(inputText)
            val outputText = outputFormat.format(date!!)
            publishedetails!!.text = outputText
        } catch (ex: Exception) {
            Log.i("Exception", "------------> " + ex.message)
        }
    }

    private fun cliqueImagem(result: Result?) {
        imageDetails!!.setOnClickListener {
            val intent = Intent(this@DetailsActivity, DetailsImageActivity::class.java)
            intent.putExtra(DETAILSIMAGE_KEY, result)
            startActivity(intent)
        }
    }

    override fun Onclick(result: Result?) {
        val intent = Intent(this, DetailsImageActivity::class.java)
        val bundle = Bundle()
        bundle.putParcelable(MainActivity.RESULT_KEY, result)
        intent.putExtras(bundle)
    }


    companion object {
        const val DETAILSIMAGE_KEY = "imagedetails"
    }

}